hal/halx86/apic/apictimer.c

```diff
@@ -59,26 +65,84 @@ ApicInitializeTimer(ULONG Cpu)
 
 /* PUBLIC FUNCTIONS ***********************************************************/
 
+VOID 
+NTAPI
+HalInitializeProfiling(VOID)
+{
+    KeGetPcr()->HalReserved[HAL_PROFILING_INTERVAL] = HalCurProfileInterval;
+    KeGetPcr()->HalReserved[HAL_PROFILING_MULTIPLIER] = 1; /* TODO: HACK */
+}
+
 VOID
 NTAPI
 HalStartProfileInterrupt(IN KPROFILE_SOURCE ProfileSource)
 {
-    UNIMPLEMENTED;
-    return;
+    LVT_REGISTER LvtEntry;
+
+    /* Only handle ProfileTime */
+    if (ProfileSource == ProfileTime)
+    {
+        /* OK, we are profiling now */
+        HalIsProfiling = TRUE;
+
+        /* Set interrupt interval */
+        ApicWrite(APIC_TICR, KeGetPcr()->HalReserved[HAL_PROFILING_INTERVAL]);
+
+        /* Unmask it */
+        LvtEntry.Long = 0;
+        LvtEntry.TimerMode = 1;
+        LvtEntry.Vector = APIC_PROFILE_VECTOR;
+        LvtEntry.Mask = 0;
+        ApicWrite(APIC_TMRLVTR, LvtEntry.Long);
+    }
```

Comment:
```
else
{
    UNIMPLEMENTED;
}
```

Or does (checked) WS03sp2 actually just ignores (= no error, no nothing) other values?

Reply:
It doesn't ignore it completely; it just calls another function that unconditionally returns error code 0xC00000BB, that isn't used. I think that's unnecessary to implement.

Reply:
Maybe add that explanation as a comment then? (Both places.)

Reply:
OK, expanded comment